### DISCUSSION TITLE
Unable to access Rest API even after following all rules

### DISCUSSION THREAD

**Discussion by lexy10** [ant-media/Ant-Media-Server#6639] [Q&A] [answered]
Please can anyone point me to how else I can connect to the rest API of antmedia. I am using docker and I have updated the conf/red5.properties file with 

server.jwtServerControlEnabled=true
server.jwtServerSecretKey={{MySecretKey}}

I have also generated the jwt token and using it as 

curl -X GET -H "Content-Type: application/json" -H "ProxyAuthorization:eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.tA6sZwz_MvD9Nocf3Xv_DXhJaeTNgfsHPlg3RHEoZRk" "https://example.com:5443/rest/v2/system-resources"

I have restarted too but I still get forbidden. What else do I do?

**Comment by lastpeony** [comment, marked as answer]
First go through this again.
https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/management-rest-apis/#jwt-token-authentication

Make sure that secret key is 32 characters.
Be sure you generate JWT as stated in documentation screen shot.
![image](https://github.com/user-attachments/assets/d4c21988-b48e-4a3c-b212-935457a6afee)

`curl -X GET -H "Content-Type: application/json" -H "ProxyAuthorization:eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.tA6sZwz_MvD9Nocf3Xv_DXhJaeTNgfsHPlg3RHEoZRk" "https://example.com:5443/rest/v2/system-resources"`





Also,
Can you try to authenticate with username password?

curl --location 'http://localhost:5080/rest/v2/users/authenticate' \
--header 'Content-Type: application/json' \
--data '{ "email": "username", "password": "your-md5-webpanel-password" }'

After success you can use the cookie to make a call to "https://example.com:5443/rest/v2/system-resources"

take a look at this:
https://antmedia.io/how-to-make-secure-ant-media-api-requests-to-web-panel/#step-1-md-5-hash-of-your-password

**Comment by lexy10** [comment]
What I did was, after setting it up inside red5.properties, i now came to untick jwt for IP and enabled jwt for Rest and then entered the secret key inside the settings page for the application on my web panel, now it works... thank you.
